WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java

```diff
@@ -128,18 +136,20 @@ public int onStartCommand(Intent intent, int flags, int startId) {
     }
 
     private void stopRefresh() {
-        this.mServiceBlogId = null;
-        this.mServiceRequestedTimeframe = StatsTimeframe.DAY;
-        this.mServiceRequestedDate = null;
-        this.mServiceStartId = 0;
-        for (Request<JSONObject> req : statsNetworkRequests) {
-            if (req != null && !req.hasHadResponseDelivered() && !req.isCanceled()) {
-                req.cancel();
+        synchronized (statsNetworkRequests) {
```

Comment:
We should rename `statsNetworkRequests` to `mStatsNetworkRequests` - I was confused while reading the diff, I thought it was loca (so no reason to synchronize it).